htlcswitch/test_utils.go

```diff
@@ -1335,3 +1377,49 @@ func timeout(t *testing.T) func() {
 		close(done)
 	}
 }
+
+// WaitPredicate is a helper test function that will wait for a timeout period
+// of time until the passed predicate returns true. This function is helpful as
+// timing doesn't always line up well when running integration tests with
+// several running lnd nodes. This function gives callers a way to assert that
+// some property is upheld within a particular time frame.
+func WaitPredicate(pred func() bool, timeout time.Duration) error {
```

Comment:
`lntest/testutils`? 🤔

Reply:
import cycle because both packages import `routing`. yes this really should go in it's own utils package with no deps

Reply:
done